p2p/transport/websocket/addrs.go

```diff
@@ -10,52 +10,28 @@ import (
 	manet "github.com/multiformats/go-multiaddr/net"
 )
 
-// Addr is an implementation of net.Addr for WebSocket.
-type Addr struct {
+// addrWrapper is an implementation of net.Addr for WebSocket.
+type addrWrapper struct {
 	*url.URL
 }
 
-var _ net.Addr = (*Addr)(nil)
+var _ net.Addr = addrWrapper{}
 
 // Network returns the network type for a WebSocket, "websocket".
-func (addr *Addr) Network() string {
+func (a addrWrapper) Network() string {
```

Comment:
why not  a *addrWrapper ?

Reply:
Why would it? This is faster without a pointer dereference.

Reply:
I know I am being pedentic, but `a` / `addr` is not used, this LITERALLY does nothing:
```x86asm
        TEXT    "".addrWrapper.Value(SB), NOSPLIT|ABIInternal, $0-8
        LEAQ    go.string."websocket"(SB), AX
        MOVL    $9, BX
        RET

        TEXT    "".(*addrWrapper).Pointer(SB), NOSPLIT|ABIInternal, $0-8
        LEAQ    go.string."websocket"(SB), AX
        MOVL    $9, BX
        RET
```
Both functions literally have the same assembly.
Here value is correct but this just because you are not modifying the `addrWrapper`. Even if you did, a pointer like this should cost no more than 3ns. Caring about performance here does not make sense.

Rules I personally follow:
- First start by using value receiver, this is to be clear that your methods are read only.
- However if you need shared state then use pointers on all methods (to get around promotion issues).

Reply:
The only thing is it was a pointer shouldn't we leave it a pointer ?

Reply:
> The only thing is it was a pointer shouldn't we leave it a pointer ?

Why ? It's readonly.